blob/src/main/java/com/bazaarvoice/emodb/blob/db/s3/S3RateLimiter.java

```diff
@@ -0,0 +1,151 @@
+package com.bazaarvoice.emodb.blob.db.s3;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.AmazonS3Exception;
+import com.google.common.base.Throwables;
+import com.google.common.reflect.AbstractInvocationHandler;
+import com.google.common.reflect.Reflection;
+import com.google.common.util.concurrent.RateLimiter;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+
+public class S3RateLimiter {
```

Comment:
In future, We got to move this to a common module and have it shared across incremental stash, here and else where.....

Reply:
agree, or we can try to find some external library, but it's out of scope of this PR